python/ray/serve/scripts.py

```diff
@@ -232,49 +234,170 @@ def start(
+
+        if import_path.count(":") != 1:
+            raise click.ClickException(
+                "Import path must be of the form 'module.submodule:app_or_builder', "
+                f"got: '{import_path}'."
+            )
+        app = ServeApplicationSchema(
+            import_path=import_path,
+            runtime_env=runtime_env,
+            args=arguments,
+        )
+        if name is not None:
+            app.name = name
+        config = ServeDeploySchema(applications=[app])
+
+    assert isinstance(config, ServeDeploySchema)
+    return config
+
+
 @cli.command(
-    short_help="Deploy Serve applications from a YAML config file.",
+    short_help="Deploy an application or group of applications.",
     help=(
-        "This supported config format is ServeDeploySchema. "
-        "This call is async; a successful response only indicates that the "
-        "request was sent to the Ray cluster successfully. It does not mean "
-        "the the deployments have been deployed/updated.\n\n"
-        "Existing deployments with no code changes will not be redeployed.\n\n"
-        "Use `serve config` to fetch the current config(s) and `serve status` to "
-        "check the status of the application(s) and deployments after deploying."
+        "Deploy an application from an import path (e.g., main:app) "
+        "or a group of applications from a YAML config file.\n\n"
+        "Passed import paths must point to an Application object or "
+        "a function that returns one. If a function is used, arguments can be "
+        "passed to it in 'key=val' format after the import path, for example:\n\n"
+        "serve deploy main:app model_path='/path/to/model.pkl' num_replicas=5\n\n"
+        "This command supports different 'providers'. By default, it uses a 'local'"
+        "provider that makes a REST API request to a running Ray cluster. "
+        "Not all arguments are supported by all providers."
+    ),
+)
+@click.argument("config_or_import_path")
+@click.argument("arguments", nargs=-1, required=False)
+@click.option(
+    "--runtime-env",
+    type=str,
+    default=None,
+    required=False,
+    help="Path to a local YAML file containing a runtime_env definition.",
+)
+@click.option(
+    "--runtime-env-json",
+    type=str,
+    default=None,
+    required=False,
+    help="JSON-serialized runtime_env dictionary.",
+)
+@click.option(
+    "--working-dir",
+    type=str,
+    default=None,
+    required=False,
+    help=(
+        "Directory containing files that your application(s) will run in. When using "
+        "the 'local' provider, this must be a remote URI to a .zip file (e.g., S3 "
+        "bucket). Other providers may support local paths. This overrides the "
+        "working_dir in --runtime-env if both are specified."
+    ),
+)
+@click.option(
+    "--name",
+    required=False,
+    default=None,
+    type=str,
+    help="Custom name for the application(s).",
```

Comment:
How does this setting work for multi-app? Can you specify multiple names?

Reply:
it's one name for the set of applications (service in anyscale terms)

Reply:
Synced offline; we'll keep this setting to maintain parity with `serve run` and support applications that use `serve.get_app_handle()`.